orttraining/orttraining/test/training_api/common/synthetic_data_loader.h

```diff
@@ -15,31 +15,23 @@
 
 #include <memory>
 #include <utility>
+#include <variant>
 #include <vector>
 
 namespace onnxruntime {
 namespace training {
 namespace test {
 namespace training_api {
 
-template <typename T>
-struct TypedSyntheticInput;
+typedef std::vector<std::variant<int32_t, int64_t, float, uint8_t>> SyntheticDataVector;
 
-struct SyntheticInput {
-  explicit SyntheticInput(const std::vector<int64_t>& shape) : shape_(shape) {
```

Comment:
>onst std::vector<int64_t>& sha [](http://example.com/codeflow?start=27&length=30)

See coding standards.
Use TensorShapeVector for shapes.
Use gsl::span<const T> for input parameters.
Use InlinedVector<T> for other instances of std::vector